feat(cms): converting the site to be a cms vs static site

jsundquist opened the pull request:
Take a look over the files and let me know before we merge it in. Let me know if there is anything that you think needs to be changed. Note that you need to create an `.env` file locally with the items within the `.env-template`. The same environment variables would need to be created on AWS as well.

jsundquist committed:
feat(cms): converting the site to be a cms vs static site

jsundquist committed:
fix(env): removing the .env file in favor of a template to be used instead

mwillbanks commented on models/Sponsor.js:
```diff
@@ -0,0 +1,37 @@
+	website: {type: String},
+	facebook: {type: String},
+	github: {type: String},
+	googleplus: {type: String},
```
G+ is dead :)

jsundquist replied:
True, this was mainly coming from last years sponsors and the social media locations they provided us.

mwillbanks commented on .env-template:
```diff
@@ -0,0 +1,3 @@
+COOKIE_SECRET={{cookieSecret}}
+MONGO_URI={{mongoUri}}
+
```
Add:
`CLOUDINARY_URL={{cloudinaryUrl}}`

jsundquist replied:
Right now the site is setup to use local file storage vs uploading the file to cloudinary. Do we want to setup a cloudinary account to upload images to?

mwillbanks replied:
I don't actually, it was required when I setup the instance locally, we can completely disable it by removing the mention of them from the models.

mwillbanks replied:
Example:
```
  heroImage: { type: Types.CloudinaryImage },
  images: { type: Types.CloudinaryImages },
``

Will cause keystone to throw an error if the URL is not defined.

jsundquist replied:
Removed the gallery model and pages as we are not using this functionality now or possibly in the future.

mwillbanks commented on .editorconfig:
```diff
@@ -0,0 +1,14 @@
+charset = utf-8
+trim_trailing_whitespace = false
+insert_final_newline = true
+indent_style = tab
```
indent_style conflicts with the airbrb styleguide which you have for eslint, please change to spaces.
https://github.com/airbnb/javascript#whitespace

jsundquist replied:
Updated to be indent_style=space to match the airbnb styles

mwillbanks commented on keystone.js:
```diff
@@ -0,0 +1,65 @@
+// and documentation.
+
+keystone.init({
+	'name': 'Midwest PHP',
```
Midwest PHP 2019

jsundquist replied:
Updated to be Midwest PHP 2019

mwillbanks commented on keystone.js:
```diff
@@ -0,0 +1,65 @@
+	'name': 'Midwest PHP',
+	'brand': 'Midwest PHP',
+
+	// 'sass': 'public',
```
Do you plan to keep the sass or not?  If so then the commented out stuff is fine but otherwise remove.

jsundquist replied:
Nope, will yank that.

jsundquist replied:
Removed the commented out code

mwillbanks commented on keystone.js:
```diff
@@ -0,0 +1,65 @@
+// Load your project's Models
+keystone.import('models');
+
+keystone.mongoose.set('debug', true);
```
Set this based on the environmental .env configuration.

jsundquist replied:
Removed the code as it was only being used when I was testing locally previously.

mwillbanks reviewed (changes requested):
A few larger items:
1. Remove console.log statements.
2. Provide a README.md with how to get started.
3. The editor config and styleguide do not match, we should fix that before we proceed.
4. Use const/let vs. var.

mwillbanks commented:
FYI, I'll issue a PR to this branch with a lot of the eslint fixes.

mwillbanks committed:
Fix ESLint Errors

* Some of this required rewriting certain areas of functions, mainly
everything should be idential to where it was before.

jsundquist committed:
Merge pull request #6 from MidwestPHP/fix-eslint-errors

Fix ESLint Errors

jsundquist committed:
feat(contact): updated the contact form to look more like the site

jsundquist committed:
Merge branch 'feat/cms/switch-static-to-cms' of github.com:MidwestPHP/midwestphp-2019 into feat/cms/switch-static-to-cms

jsundquist committed:
feat(gallery): removing the gallery as we are not making use of it

jsundquist committed:
fix(format): replacing all tabs with spaces

jsundquist committed:
fix(config): updating editor config to match airbnb config

jsundquist committed:
fix(config): updated the site name to reflect the time it is for

jsundquist committed:
fix(favicon): updating to use the midwest php fav icon vs keystone

jsundquist committed:
feat(analytics): adding in the analytics code

jsundquist committed:
fix(registration): updated to have the latest information

jsundquist committed:
fix(favicon): updated to have the correct favicons for all platforms

jsundquist committed:
fix(formatting): replaced tabs with spaces

jsundquist committed:
feat(readme): adding in the missing readme file

jsundquist commented:
> Remove console.log statements.

Only console log item was within the update script. Leaving this one in.

> Provide a README.md with how to get started.

Added in the README.md

> The editor config and styleguide do not match, we should fix that before we proceed.

Updated

> Use const/let vs. var.

Updated in your pull request
